src/domain/models/bertcombisum.py

```diff
@@ -0,0 +1,242 @@
+        self.hidden_dim = hparams.hidden_dim
+        self.decoder_dim = hparams.decoder_dim
+        self.learning_rate = hparams.learning_rate
+        self.n_sents_per_summary = hparams.n_sents_per_summary
+        self.c_puct = hparams.c_puct
+        self.ucb_sampling = hparams.ucb_sampling
+        self.weight_decay = hparams.weight_decay
+        self.batch_idx = 0
+        self.criterion = torch.nn.BCEWithLogitsLoss(reduction="none")
+        self.idxs_repart = torch.zeros(
+            50, dtype=torch.float32, device=self.tensor_device
+        )
+        self.test_size = len(self.splits["test"])
+        self.targets_repart = torch.zeros(
+            50, dtype=torch.float64, device=self.tensor_device
+        )
+        self.train_size = len(self.splits["train"])
+        self.my_core_model = Summarizer(self.tensor_device)
+        if hparams.n_jobs_for_mcts == -1:
+            self.n_processes = os.cpu_count()
+        else:
+            self.n_processes = hparams.n_jobs_for_mcts
+        self.pool = mp.Pool(processes=self.n_processes)
+
+    def __my_document_level_encoding(self, contents):
+        '''
+        input:
+            contents : list of output berttokenizer
+        '''
+        sent_scores, mask_cls = self.my_core_model(contents)
+        return sent_scores, mask_cls
+
+    def forward(self, batch, subset):  # (data_batch, "train" or "test")
+        ids, contents, abstracts, raw_contents, raw_abstracts, scorers = batch
+        batch_size = len(contents)
+
+        contents_extracted, valid_sentences = self.__my_document_level_encoding(contents)
+
+        _, greedy_idxs = torch.topk(
+            contents_extracted, self.n_sents_per_summary, sorted=False)
+
+        if subset == "train":
+            greedy_rewards = []
+            for scorer, sent_idxs in zip(scorers, greedy_idxs):
+                greedy_rewards.append(scorer(sent_idxs.tolist()))
+            greedy_rewards = torch.tensor(greedy_rewards)  if isinstance(greedy_rewards[0], list) else torch.tensor([greedy_rewards])
+
+            ucb_results = self.pool.map(
+                UCBProcess(self.ucb_sampling, self.c_puct), scorers,
+            )
+
+            ucb_targets = torch.tensor(
+                [r[0] for r in ucb_results], device=valid_sentences.device
+            )
+            ucb_deltas = torch.tensor([r[1] for r in ucb_results])
+
+            #Padding if max len_doc < 50 sentences
+            pad_ = torch.zeros((ucb_targets.size(0), ucb_targets.size(1) - valid_sentences.size(1)), dtype=torch.bool).to(ucb_targets.device)
+            valid_sentences = torch.cat((valid_sentences, pad_), dim=-1).to(ucb_targets.device)
+
+            target_distro = 10 ** (-10 * (1 - ucb_targets)) * valid_sentences
+            pad_ = torch.zeros((target_distro.size(0), target_distro.size(1) - contents_extracted.size(1))).to(target_distro.device)
+            contents_extracted = torch.cat((contents_extracted, pad_), dim=-1).to(target_distro.device)
```

Comment:
Par exemple ici, cat() ne permet pas de passer une device, mais va planter si les deux tenseurs ne sont pas déjà sur la même device. Comme tu sais déjà plus haut qu'ils sont les deux sur GPU, pas besoin d'appeler to().

N.B. Appeler .to('cuda') sur un tenseur déjà sur GPU est un _no-call_, alors on ne perd pas de temps, mais c'est une bonne pratique d'apprendre à éliminer les .to()